### ISSUE TITLE
Run arithmetic fuzzer tests in CI

### ISSUE THREAD

**Issue by ggwpez** [paritytech/substrate#13668]
The [fuzzy tests](https://github.com/paritytech/substrate/blob/b9120f8c041718a7fbed97a302d5af71bfe2048c/primitives/arithmetic/fuzzer/src/per_thing_mult_fraction.rs#L30) in `sp-arithmetic` are currently not running in any CI.

Example of running them with [honggfuzz](https://github.com/rust-fuzz/honggfuzz-rs):
```bash
cd primitives/arithmetic/fuzzer/
# Run for 10 secs
HFUZZ_RUN_ARGS="-t 10 --exit_upon_crash" cargo hfuzz run multiply_by_rational_with_rounding
# Repeat for all files in that folder, only `normalize.rs` looks broken.
```
`hfuzz_workspace/` should be uploaded in case it fails.

(You can move this issue to this CI/CD repo if you think it fits better)

**Comment by altaua** [comment]
fwiw `honggfuzz -t 10` doesn't actually limit the total runtime to 10sec, just the runtime per individual test case. To make honggfuzz exit eventually rather than running forever, a `--run_time` or `--iterations` limit is needed.

How long would we want the fuzzing runs to be? My intuition is that fuzzing mostly makes sense if you run the fuzzer for a good looong time, on the order of hours, which would make it prohibitively expensive to perform as part of regular per-commit ci checks.

**Comment by ggwpez** [comment]
>fwiw honggfuzz -t 10 doesn't actually limit the total runtime to 10sec, just the runtime per individual test case. To make honggfuzz exit eventually rather than running forever, a --run_time or --iterations limit is needed.

Okay thanks!

>My intuition is that fuzzing mostly makes sense if you run the fuzzer for a good looong time, on the order of hours, which would make it prohibitively expensive to perform as part of regular per-commit ci checks.

Yes it is definitely a trade-off. Current situation: never run them and they trivially fail after a few seconds.  
So if we run each 10 seconds in the CI, over time it will amount to a long time since we have a lot or MRs. It is surely worse than running them for hours, but in these cases the functions that we test are not *that* complicated. Basically some math functions.  
Therefore I think even running in the CI each time for 10 secs is a good start. Developers will hopefully run them longer locally when they change these functions :see_no_evil:.

**Comment by altaua** [comment]
Good point! I'll look into adding some short fuzzing runs to the master & mr pipelines then.

**Comment by ggwpez** [comment]
Thanks. On the current master branch at least one of them should fail and on this branch they should all be green https://github.com/paritytech/substrate/pull/13660/.  
Just for your debugging :smile:
